client/client-multi/client-c/CMakeLists.txt

```diff
@@ -261,3 +290,11 @@ install(TARGETS kaac DESTINATION lib)
 add_subdirectory(Modules/doxygen)
 
 add_subdirectory(Modules/cppcheck)
+
+externalproject_add(kaa_rsa_keygen SOURCE_DIR "${CMAKE_CURRENT_LIST_DIR}/tools/kaa_encryption"
```

Comment:
I think it is not possible to add the dependence via execute_process(). Is there a way to guarantee that the build process will go in the following order if execute_process():

1. Keys generator is built before SDK is built
2. Keys are generated before SDK is built.
3. Generated header is copied before SDK is built.

AFAIK,  if [execute_process()](https://cmake.org/cmake/help/v3.0/command/execute_process.html) is used it executes the given command *prior the build system generation*. But the keys are generated at the build time. So there is no way to do it the way you described. Correct me please if I'm wrong.
P.S. Are there any troubles with `externalproject_*`?